scripts/irods/test/test_dynamic_peps.py

```diff
@@ -492,3 +492,59 @@ def test_if_Genquery2Input_is_exposed__issue_7570(self):
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_1', self.user.zone_name])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_2', '0'])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_3', '0'])
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-irods_rule_language' or test.settings.RUN_IN_TOPOLOGY, "Requires NREP and single node zone.")
+    def test_if_StructFileExtAndRegInp_is_exposed__issue_7413(self):
+        with temporary_core_file() as core:
+            avu_prefix = 'test_if_StructFileExtAndRegInp_is_exposed__issue_7413'
+
+            # Add a rule to core.re which when triggered will add AVUs to the user's
+            # session collection.
+            core.add_rule(dedent('''
+            pep_api_struct_file_ext_and_reg_post(*INSTANCE, *COMM, *STRUCTINP) {{
+                writeLine("serverLog",*STRUCTINP);
+                *v = *STRUCTINP.objPath;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_0', *v, '');
+
+                *v = *STRUCTINP.collection;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_1', *v, '');
+
+                *w = *STRUCTINP.oprType;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_2', *w, '');
```

Comment:
No `.flags` or keyValPair_t AVUs?

Reply:
Did `.flags` and one of the KVPs-- is that ok, or should I try to hit both KVPs?

Reply:
Looks like you've exercised all five parts of the struct.   I think you're good to go.

What is the 'other' KVP you're referencing in your comment?

Reply:
I agree. The test looks complete to me.

I second @trel's question.

Reply:
Sorry, I meant another element in the KVP. If just one is fine, then I think we're good.

Reply:
Let's include one more element in the KVP so we're absolutely sure things are working.

If it's not clear which one to include, try adding `-f` to the `ibun` call and seeing if the force flag appears in the KVP. The force flag should appear as `"forceFlag"`.

https://github.com/irods/irods/blob/f51b33a1f83ac612536e2b6093209367f79dd2ed/lib/core/include/irods/rodsKeyWdDef.h#L11

Which is used in bunUtil.cpp _(i.e. the workhorse of `ibun`)_.

https://github.com/irods/irods/blob/f51b33a1f83ac612536e2b6093209367f79dd2ed/lib/core/src/bunUtil.cpp#L153-L155